lib/fog/azurerm/storage.rb

```diff
@@ -16,31 +16,41 @@ class AzureRM < Fog::Service
       recognizes :debug
 
       request_path 'fog/azurerm/requests/storage'
+      # Azure Storage Account requests
       request :create_storage_account
       request :list_storage_accounts
       request :delete_storage_account
       request :list_storage_account_for_rg
       request :check_storage_account_name_availability
       request :get_storage_access_keys
+      # Azure Storage Disk requests
       request :delete_disk
-      request :get_blob_metadata
-      request :get_container_metadata
-      request :set_blob_metadata
-      request :set_container_metadata
+      # Azure Storage Container requests
       request :create_container
       request :delete_container
       request :list_containers
+      request :get_container_metadata
+      request :set_container_metadata
       request :get_container_properties
       request :get_container_access_control_list
+      # Azure Storage Blob requests
+      request :list_blobs
+      request :set_blob_metadata
+      request :get_blob_metadata
+      request :set_blob_properties
+      request :get_blob_properties
+      request :upload_block_blob_from_file
+      request :download_blob_to_file
+      request :delete_blob
 
       model_path 'fog/azurerm/models/storage'
       model :storage_account
       collection :storage_accounts
       model :data_disk
-      model :container
-      collection :containers
-      model :blob
-      collection :blobs
+      model :directory
+      collection :directories
+      model :file
```

Comment:
Quoted the reply here:
>Yes, Blobs and Files are different in Azure Storage. However, there is no File service API supported in the azure-storage gem for now. So the best mapping for the directory and file in the Fog is container and blob. We may consider switch to the File APIs when they are ready in the azure-storage gem, otherwise we need call with REST APIs for File directly.